### ISSUE TITLE
Navit download of maps

### ISSUE THREAD

**Issue by mensch21** [navit-gps/navit#866]
Hello Navit team,

The download of maps is unfortunately limited to 3.8 GB may be based on FAT32 file system.
If maps are loaded on internal storage, for example Android extended file system, the size of a file is nearly not limited.

Could'nt you check the file system and free space before loading down and give a release if the requirements are complied?

Then it will be possible to load down Europe map again. I tried to load down europe map on PlanetExtractor but at least it will take two days (on Windows PC). The internal download on Android tablet is quite fast (I think it is not XML based).

Best regards,
Joachim Brachvogel (mensch21)

**Comment by jandegr** [comment]
Hi,
Can you tell the android version and make of your device ?
The download limit has been there all along, so I suppose downloading europe on android has never worked, but even if downloading all of europe is not an everyday usecase it would better get fixed because the predefined region of France will soon hit the downloadlimit as well.

Have you ever managed to get a single map for europe working on android in another way ?

thx.

**Comment by mensch21** [comment]
Hi,

 

The android version of my device is 7.1.1.

 

My Europe map is 1 or 2 years old and quite often I was off road in

the last time.

So I tried to get the latest europe map directly by navit software

on this tablet. Two years is not an every day usecase - that's wright.

 

One or two years ago I was successful loading europe map from Planet Extractor

(http://maps3.navit-project.org/) on a windows-pc.

 

Yes, it took a long time but at least I was successful and in 2 or 3 times

I managed to copiy this 12GB file to the android tablet by network. The tablet has

an internal storage of 64GB. So there is enough space.

 

Now I loaded single countries by navit software. But unfortunately the more maps

I have active in navit the slower navit works. When I rename all the

maps I do not need at the moment (ren country.bin country.binx),

and at least I keep Germany.bin as active map (if I am in this country),

then navit is much faster.

 

The same I do since years with my 25000 - 30000 POIs (the most are

icons for places where I can stay overnight with my camper). These POIs are

filed in countries. If I activite all off the POIs navit stands more or less still.

 

I will use the country maps in future. I wrote a software (name is MyPoisActive).

With this app I turn on and off the countries I do not need now (rename x.poi x.poix

and back)

 

I will advance this MyPoisActive app even by renaming the country maps too using a small helpfile:

Country (POIs)     Map

Deutschland        Germany.bin

Niedrlande          BENELUX.bin

Belgien                BENELUX.bin

aso...

This helps managing 1-> n relations.

 

I added a screenshot of my navit environment. Only with this Navit software I can show all

my pois (icons) on the map like in the screenshot.

 

Even I created PDF files from the places containing descriptions of the places and some pictures.

So from a distance I can take a look for the nicest places I like to stay overnight.

For example "D/3202" is the link to the PDF file. I wrote a app "MyPoisShowPdf"

I can select the country (D for Deutschland) and enter the number (for example: 3202" and

press "Show PDF" button.

 

The kings way would be... navit enables a long click on the map and starts a

user defined app with the north and east coordinates. Then I could start the PDF viewer

more or less directly from navit. And with a toggle button I could switch between "set routing

position" and the user defined app. But this is a dream :-)

 

I use Navit since 2013 and I am very satified with it.

As a retired programmer I like to keep my brain active - and it works.

 

With my best regards

Achim Brachvogel (mensch21)

 

 

 

 

 

 

 

 

 

 

 

 

 

 

 

 

Gesendet: Freitag, 06. September 2019 um 12:09 Uhr
Von: "jandegr" <notifications@github.com>
An: navit-gps/navit <navit@noreply.github.com>
Cc: mensch21 <joachim.brachvogel@gmx.de>, Author <author@noreply.github.com>
Betreff: Re: [navit-gps/navit] Navit download of maps (#866)


Hi,
Can you tell the android version and make of your device ?
The download limit has been there all along, so I suppose downloading europe on android has never worked, but even if downloading all of europe is not an everyday usecase it would better get fixed because the predefined region of France will soon hit the downloadlimit as well.

Have you ever managed to get a single map for europe working on android in another way ?

thx.

**Comment by jandegr** [comment]
Hi,
I just built a testversion with the download limit removed. I then started downloading North-America, it is 6623MB, also well above the 4GB limit so I think the test is valid but it takes only about half the time of downloading Europe. 

Result : downladed just fine, noticed no problems and could plan a route of 4980km from the SouthEast to the Westcoast.
This was just a quick test but the result is encouraging to investigate a little further.
Tested on 32 bit, so on the 64 bit version it will work just as well. 
If you're interested I can provide you with a testversion so you can try for yourself.

A note on downloading individual maps 
There is quite a bit of overlap in the predefined regions, they are just rectangles after all.
If you download a handfull you may find they take up more storage than all of europe in one binfile.

You mention you added a screenshot but that does not show up here on github.

**Comment by lains** [comment]
> The kings way would be... navit enables a long click on the map and starts a
> user defined app with the north and east coordinates. Then I could start the PDF viewer
> more or less directly from navit. And with a toggle button I could switch between "set routing
> position" and the user defined app. But this is a dream :-)

If you're looking for having an implementation of this use case in Android, this should be possible once https://github.com/navit-gps/navit/pull/812 is merged.
You target third-party app should then be able to receive a geo intent from Navit.

**Comment by mensch21** [comment]
Hi,

 

Many thanks for your effort.

My incident I left on github was just a hint how it could be possible

to load dow bigger files. It should no be limited on FAT32 if the file

are loaded down on extended file systems for example.

 

For two years I was satisfied with europe map, exceopt some big

blue squares for example in south of France and Sweden. I was there

in "blue motion" :-) There is something with rivers or lakes, perhaps

"open chains"... I took a look on the edges of the squares but

I could not make out what it is.

Even on the latest doenload form navit on my tablet I got

blue sqares, I think it was in Litauen... but this does not matter...

Some days ago in France the blue square was about 100 square Km - so,

a bit bigger... (between Nizza and Grenoble)

 

Testversions are a problem... Two years ago, I think the 3rd of December it

was, I informed the navit team that the latest update had no icons

in the .APK. And the APK had half of the size as usual.

It was this "drawable-nodpi-v4" directory the APK did not contain.

This copy statement was missing in you new built file -it  has'nt been taken over...

I installed the navit version before this "gap", but over night I got back this "smaller"

navit release, because wireless network at home...

 

When I came back from a trip, somedays later, there was a small revolt on google.plays

rewievs on navit. But, so you can see that this software is not just loaded down, tested

and removed - no, it is really used.

 

So a test version may be updated automaically.

Today, it was a rainy day, I advanced my software "MyPoisActive"

managing the maps to by renaming the extention (add a "x" or remove it).

Now it works. (But with android studio I get crasy, I think I was more a

programmer for open-VMS or Windows or Linux. C, C++ or C# or even Fortran

is OK, but Java for Android is awful :-)

 

Yes the overlap whit the rectangles it a problem. There is a lot of redundant

data, redarding whole europe splitted in country files.

 

Before two years ago and now I loaded more than a handful of country maps.
Since I stopped working I go though europe and have all of the maps of the

countries on my tablet. At the moment I drive about 50000 Km a year.

And sometime I come back home an do a bit programming... :-)

 

I added a screenshot as a attachment in my mail I answered from.

If you did not get this screenshot and you are interested in, I will send

you this screenshot in a method "you like to have".

Even I could send you other screenshots, perhaps from my "NavitPoisActive"

or "MyPoisShowPdf", all for Navit App.

 

Best regards and have a nice evening

Achim Brachvogel (mensch21)

 

 

 

 

 

 

 

 

 

 

 

 

 

Gesendet: Freitag, 06. September 2019 um 21:08 Uhr
Von: "jandegr" <notifications@github.com>
An: navit-gps/navit <navit@noreply.github.com>
Cc: mensch21 <joachim.brachvogel@gmx.de>, Author <author@noreply.github.com>
Betreff: Re: [navit-gps/navit] Navit download of maps (#866)


Hi,
I just built a testversion with the download limit removed. I then started downloading North-America, it is 6623MB, also well above the 4GB limit so I think the test is valid but it takes only about half the time of downloading Europe.

Result : downladed just fine, noticed no problems and could plan a route of 4980km from the SouthEast to the Westcoast.
This was just a quick test but the result is encouraging to investigate a little further.
Tested on 32 bit, so on the 64 bit version it will work just as well.
If you're interested I can provide you with a testversion so you can try for yourself.

A note on downloading individual maps
There is quite a bit of overlap in the predefined regions, they are just rectangles after all.
If you download a handfull you may find they take up more storage than all of europe in one binfile.

You mention you added a screenshot but that does not show up here on github.

**Comment by mensch21** [comment]
Hello "lains",

 

Perhaps a dream comes true... :-)

 

I read issue #812 but I could not make out if this merge/release will touch me.

But I am very interesting...

 

My dream is... in Navit on the map I do a long click somewhere onto the map

(where one of my Poi-icons is) and Navit starts a user defined app or tntent

with north and east coordinates as parameters (and perhaps the zoom-level too).

Then I could set a frame on these north and east values in this app and take a

look into my poi-files for matching pois. The frame is necessary because the finger

is not exact as the coordinates are... and I need a "precision landing" or I fail.


For most of my pois I have PDF-files and the information to this (a kind of link) I have

in the lables on the pois of the map (on the screen and so in my poi files too).

With this catching and matching data I could start a process (process start ...)

and show this PDF immediately.

 

When stopping this PDF I return back to Navit, may be to see the next PDF-Poi. or...

if I got a Poi I like to go to, I switch (by a toggle button) back to the original

functionalitiy in Navit "set routing", and Navit leads me to this Poi.

This is what I ment is a dream :-)

 

In some cases, this can happen using frames, I get more than one hit, I could

create a window containing these icons and labels and I can choose what I like to see.

May be I choose a place for sleeping over night and close to this place there is


a thermal bath or a ice ring or some other stuff (I have in my pois too). But this is

customizing by the user...


 



Best regards and printf("have a nice evening\r\n");

Achim Brachvogel (mensch21)

 

 

 

 

 


Gesendet: Samstag, 07. September 2019 um 07:58 Uhr
Von: "lains" <notifications@github.com>
An: navit-gps/navit <navit@noreply.github.com>
Cc: mensch21 <joachim.brachvogel@gmx.de>, Author <author@noreply.github.com>
Betreff: Re: [navit-gps/navit] Navit download of maps (#866)



The kings way would be... navit enables a long click on the map and starts a
user defined app with the north and east coordinates. Then I could start the PDF viewer
more or less directly from navit. And with a toggle button I could switch between "set routing
position" and the user defined app. But this is a dream :-)


If you're looking for having an implementation of this use case in Android, this should be possible once #812 is merged.
You target third-party app should then be able to receive a geo intent from Navit.

**Comment by lains** [comment]
> I read issue #812 but I could not make out if this merge/release will touch me.
> But I am very interesting...
>  
> My dream is... in Navit on the map I do a long click somewhere onto the map
> (where one of my Poi-icons is) and Navit starts a user defined app or tntent
> with north and east coordinates as parameters (and perhaps the zoom-level too).
> Then I could set a frame on these north and east values in this app and take a
> look into my poi-files for matching pois. The frame is necessary because the finger
> is not exact as the coordinates are... and I need a "precision landing" or I fail.

What PR #812 proposes is:
When you do a long press on a specific point on the map (in Android), a contextual menul pops up (it does this already on current navit).
Today, this allows you to select one action to set that point as destination.
This PR adds one action called "View".
When pressing View, navit will create an Android geo: intent containing the longitude and latitude of the clicked point on the map. The rest is handled by Android itself (not navit anymore), and if there are apps that are able to process geo intents installed on your device (for example Google Maps), Android will propose you to open the specific location inside this apps.
If there are several apps, you will be able to choose which one to use, thus if your app "MyPoisShowPdf" can process incoming View intents using the geo: scheme, it will appear in the selection window.

Please note that your app will be provided only with the geographical Longitude and Latitude (eg: `geo:39.561946,2.634921`, but no information about the zoom level, as I don't think this is possible in the geo: intent on Android.

**Comment by mensch21** [comment]
Hi,

 

your explanation is more clear now.


So when the release is out I can get these geo data by intent

for example: getIntent().getExtras().getString("geo:");

Something like this? This would be very helpful.

 

I can live without zoom level. This zoom level was just an idea (for myself)

to set the frame size to avoid getting too many poisitions from my poi files.

But I have no feeling how it works because I did not do this before.

An I think it takes a bit of tuning. I will feel it...

 

Many thanks. I'm looking forward for the release.

 

Best regards

Achim Brachvogel (mensch21)

 

 

 

Gesendet: Sonntag, 08. September 2019 um 10:04 Uhr
Von: "lains" <notifications@github.com>
An: navit-gps/navit <navit@noreply.github.com>
Cc: mensch21 <joachim.brachvogel@gmx.de>, Author <author@noreply.github.com>
Betreff: Re: [navit-gps/navit] Navit download of maps (#866)



I read issue #812 but I could not make out if this merge/release will touch me.
But I am very interesting...

My dream is... in Navit on the map I do a long click somewhere onto the map
(where one of my Poi-icons is) and Navit starts a user defined app or tntent
with north and east coordinates as parameters (and perhaps the zoom-level too).
Then I could set a frame on these north and east values in this app and take a
look into my poi-files for matching pois. The frame is necessary because the finger
is not exact as the coordinates are... and I need a "precision landing" or I fail.


What PR #812 proposes is:
When you do a long press on a specific point on the map (in Android), a contextual menul pops up (it does this already on current navit).
Today, this allows you to select one action to set that point as destination.
This PR adds one action called "View".
When pressing View, navit will create an Android geo: intent containing the longitude and latitude of the clicked point on the map. The rest is handled by Android itself (not navit anymore), and if there are apps that are able to process geo intents installed on your device (for example Google Maps), Android will propose you to open the specific location inside this apps.
If there are several apps, you will be able to choose which one to use, thus if your app "MyPoisShowPdf" can process incoming View intents using the geo: scheme, it will appear in the selection window.

Please note that your app will be provided only with the geographical Longitude and Latitude (eg: geo:39.561946,2.634921, but no information about the zoom level, as I don't think this is possible in the geo: intent on Android.

**Comment by jandegr** [comment]
Hi,

I am very much interested in https://github.com/navit-gps/navit/pull/812 as well,
but why no open another issue for it, the title of this one is about downloading maps,

Thx.

**Comment by mensch21** [comment]
Hi,

 

I'm sorry, you are absolutely right.

This topic came out of the communication and is not content.

This should not be a chat here... but those things happens.

 

I've got information about #812. #812 is on the way and that's great.

 

I think I can close this issue. Perhhaps the limit of the download size

will be checked in one of the next releases.

 

I have a work arround for this - single countries.

 

Many thanks,

Achim Brachvogel (mensch21)

 

 

Gesendet: Sonntag, 08. September 2019 um 20:02 Uhr
Von: "jandegr" <notifications@github.com>
An: navit-gps/navit <navit@noreply.github.com>
Cc: mensch21 <joachim.brachvogel@gmx.de>, Author <author@noreply.github.com>
Betreff: Re: [navit-gps/navit] Navit download of maps (#866)


Hi,

I am very much interested in #812 as well,
but why no open another issue for it, the title of this one is about downloading maps,

Thx.

**Comment by mensch21** [comment]
I have a workaround. Perhaps the space limit will be check on the next release.
We will see...
Many thanks

**Comment by lains** [comment]
> So when the release is out I can get these geo data by intent for example: getIntent().getExtras().getString("geo:"); Something like this? This would be very helpful.

@mensch21, in order to receive geo intents in your application, you can get inspired from the code in #812 because in that PR, I implemented both receiving and sending geo: intents.
Specifically, you can have a look at:
https://github.com/navit-gps/navit/blob/9eda09bbd1d01482ba3261c377cc7b6e1c208e4d/navit/android/AndroidManifest.xml#L25
https://github.com/navit-gps/navit/blob/9eda09bbd1d01482ba3261c377cc7b6e1c208e4d/navit/android/src/org/navitproject/navit/Navit.java#L521

... even though the code is not in its simplest form there because it is integrated with navit's internal attributes and representation.

**Comment by mensch21** [comment]
Hello lains,

 

Many thanks to you for your hints to the geo intent.

 

Yes, this is not the simplest way but I red the layout and the java code

and I think I will get it done.

I'll take the essences I need for reading latitude and longitude and

after I am successful I can read my poi lists geo data having a frame (minus left,

minus up, plus right, plus down) arround its geo values and look for scores. I think

it will go and I'm looking positive. It will be a bit tuning on the frame size

because of missing feeling...

 

So even I'm looking forward to the release.

 

But... may me it could bee useful (by this long click) to set what should

have been done (geo-intent or set routing) by a toggle button and setting its

click time (this could be optional). Then the user does not have to choose on the

push-up window...

At first take a look on what is shown on geo-intent, then toggle the button

and set routing to this point (point of interest).

This could have been done in the same way like turning off the menu on short click on

the map and managing the menu on a button.

 

This was very helpful because while driving I often hit the screen by accident and got

the menu in a time I needed the map, for example when approaching a crossing and

look where to go...

Sometimes it was funny... quite often I had to turn back because I was on the wrong direction.

You cannot stop the car right before a crossing and take a look on the map where to go,

especially when there is the menu on the screen an not the map. And getting the map back

will take some time. It will be very noisy behind (blowing some horns...).

 

But... let's wait and see how it feels after this version is released.

 

Best regards

Achim Brachvogel (mensch21)

 

 

 

Gesendet: Sonntag, 15. September 2019 um 08:45 Uhr
Von: "lains" <notifications@github.com>
An: navit-gps/navit <navit@noreply.github.com>
Cc: mensch21 <joachim.brachvogel@gmx.de>, Mention <mention@noreply.github.com>
Betreff: Re: [navit-gps/navit] Navit download of maps (#866)



So when the release is out I can get these geo data by intent for example: getIntent().getExtras().getString("geo:"); Something like this? This would be very helpful.


@mensch21, in order to receive geo intents in your application, you can get inspired from the code in #812 because in that PR, I implemented both receiving and sending geo: intents.
Specifically, you can have a look at:
https://github.com/navit-gps/navit/blob/9eda09bbd1d01482ba3261c377cc7b6e1c208e4d/navit/android/AndroidManifest.xml#L25
https://github.com/navit-gps/navit/blob/9eda09bbd1d01482ba3261c377cc7b6e1c208e4d/navit/android/src/org/navitproject/navit/Navit.java#L521

... even though the code is not in its simplest form there because it is integrated with navit's internal attributes and representation.

**Comment by Jadex14rthenn** [comment]
Hi guys I'm pro**

**Comment by mensch21** [comment]
HI all,
I have been waking up by Jadex14rthenn...
And I remember I was asked for screenshots I announced, but I never dropped.

So some screenshots of NAVIT working with GEO intent...

NAVIT... some POIs... zoomed in
![1 Navit some POIs](https://user-images.githubusercontent.com/43501862/108633923-460b2d80-7477-11eb-9fc0-8994c983032b.JPG)

I click in a POI (long click) and select "Ansicht" (view) an get the apps it are able to handle GEO intent. NavPDF is my app...
I click NavPDF
![2 Navit long click on a POI](https://user-images.githubusercontent.com/43501862/108634048-05f87a80-7478-11eb-9714-907e185e2403.JPG)

The PDF of this POI is shown immediately...
![3 PDF from POI](https://user-images.githubusercontent.com/43501862/108634083-3cce9080-7478-11eb-8950-bdac54cea1d6.JPG)

If I click on a cluster of POIs, in the lower left corner of screeshot 1, I show the list (in a small window) of POIs I caught...
Then I click on one POI from list, see the PDF file and get back to the list, or with back-key NAVIT gets back the focus.
![4 POI Cluster](https://user-images.githubusercontent.com/43501862/108634182-b9fa0580-7478-11eb-9677-38296d13ad32.JPG)

And this is MyPoisShowPDF I used before NAVIT GEO intent has been released.
![5 MyPoisShowPdf](https://user-images.githubusercontent.com/43501862/108634224-ee6dc180-7478-11eb-9309-20d0a0e73c59.JPG)

Why is this intent so usefull for me? I have a lot of POIs in NAVIT (about 25000). Here is a screesnhot of some POIs from Germany.
The other countries are turned off for performance.
![0 Navit many POIs](https://user-images.githubusercontent.com/43501862/108634310-56bca300-7479-11eb-829c-b3a7c02a55ea.JPG)

It works fine and I no longer have to switch between NAVIT and MyPoisShowPDF. Great, many thanks.
